build.gradle.kts

```diff
@@ -103,6 +113,32 @@ allprojects {
 
     }
 
+    pluginManager.withPlugin("io.swagger.core.v3.swagger-gradle-plugin") {
+
+        dependencies {
+            // this is used to scan the classpath and generate an openapi yaml file
+            implementation("io.swagger.core.v3:swagger-jaxrs2-jakarta:${swaggerJaxrs2Version}")
+            implementation("jakarta.ws.rs:jakarta.ws.rs-api:${rsApi}")
+        }
+// this is used to scan the classpath and generate an openapi yaml file
+        tasks.withType<io.swagger.v3.plugins.gradle.tasks.ResolveTask> {
+            outputFileName = project.name
+            outputFormat = io.swagger.v3.plugins.gradle.tasks.ResolveTask.Format.YAML
```

Comment:
Are there any reasons why you chose YAML? Most of the API documentation I've seen so far has mostly been in the JSON format. There's no problem providing this in YAML - I'm just interested to see if there are specific reasons for this.

Reply:
Could of reasons, none of which are very firm:
- merging: we need to merge the files, and I found a plugin that does this for YAML
- YAML seems a bit less verbose (no braces, no commas, etc.)
- YAML is what most of the Swagger samples I found used 

Personally I really don't have a strong opinion on this, if we can find a merging plugin I'm fine with providing JSON, or we could even transform the final YAML to JSON, thus providing both.

Reply:
Thanks for the explanation. I totally agree with you about the advantages of YAML 🙂 For me, we can leave it in YAML.